beanstalk_worker/views.py

```diff
@@ -22,3 +25,54 @@ def run_all(request):
     count = task_service.run_all()
 
     return HttpResponse(f"Ran {count} tasks")
+
+
+@csrf_exempt
+def task_launcher(request, task_name, user_name):
+    try:
+        the_user = User.objects.get(username=user_name)
+    except User.DoesNotExist:
+        return JsonResponse({"status": "fail", "error": f"User not found for {user_name}"})
+    except Exception as e:
+        return JsonResponse(
+            {"status": "fail", "error": f"Error while loading user {user_name}", "error_details": str(e)}
+        )
+
+    try:
+        import importlib
+
+        task_parts = task_name.split(".")
+        task_fn_str = task_parts.pop()
+        task_module_str = ".".join(task_parts)
```

Comment:
not important but in case you don't know, you can use rsplit for this
`task_module_str, task_fn_str = task_name.rsplit('.', 1)`

```python
In [2]: task_module, task_fn = "a.b.c".rsplit('.', 1)

In [3]: task_module, task_fn
Out[3]: ('a.b', 'c')
```